File: Java/Examples4/RegularOctahedron.java
public class RegularOctahedron implements IThreeDimensionalObject {

    public boolean isInside(double x, double y, double z) {
	if ( (x + y + z) <= 1.0 )
	    return true;
	else
	    return false;
    }

    public String getName() {
	return "正八面体";
    }

    public double getAnalyticalSolution() {
	return 4.0/3.0;
    }

}

File: Python/Examples2/regular_octahedron.py
from i3d_object import IThreeDimensionalObject
import math

class RegularOctahedron(IThreeDimensionalObject):
	
    def is_inside(self, x, y, z):
        if (math.fabs(x) + math.fabs(y) + math.fabs(z)) <= 1.0:
            return True
        else:
            return False
            
    @property
    def name(self):
        return '正八面体'  
    
    @property
    def analytical_solution(self):
        return 4.0/3.0
